### PR TITLE
Sign Up Page

### PR DESCRIPTION
**Pull request by williampj**
Adds the sign up form to the sign up page.

### CODE REVIEW

File: pages/users/signup.js [hunk 1]
```diff
@@ -1,22 +1,171 @@
...
+                  className={classes.submit}
+                  onClick={submitForm}
+                  disabled={isSubmitting}
+                >
+                  Sign Up
+                </Button>
+              </Form>
+            )}
+          </Formik>
+          <Link href="users/login">
```
Comment (andrewc910) [inline pages/users/signup.js:161]: This should be `/users/login/`. Omitting the first slash just appends to the url so this is linking to `/users/users/login` which doesn't exist.

File: pages/users/signup.js [hunk 2]
```diff
@@ -1,22 +1,171 @@
...
+              email: '',
+              password: '',
+              confirm: '',
+            }}
+            validate={(values) => handleValidate(values)}
+            onSubmit={(values, { setSubmitting }) =>
+              handleSubmit(values, setSubmitting)
+            }
+          >
+            {({ errors, handleChange, touched, submitForm, isSubmitting }) => (
```
Comment (andrewc910) [inline pages/users/signup.js:100]: I am not too familiar with Formik but why do you need the touched attribute? Looks like it's just adding more code. I don't see a reason we need to know if something has been touched. We only need the values.

Actually, upon testing you only need `submitForm, isSubmitting, values`

File: pages/users/signup.js [hunk 3]
```diff
@@ -1,22 +1,171 @@
...
+  CssBaseline,
+  Button,
+  Grid,
+  makeStyles,
+  TextField,
+  Link,
+  Paper,
+} from '@material-ui/core';
+
+import { Formik, Form } from 'formik';
```
Comment (andrewc910) [inline pages/users/signup.js:13]: You need to import `Field` from formik. 
You also need to add `import { TextField } from 'formik-material-ui';` 

Look at comment on email field for more info.

File: pages/users/signup.js [hunk 4]
```diff
@@ -1,22 +1,171 @@
...
+            validate={(values) => handleValidate(values)}
+            onSubmit={(values, { setSubmitting }) =>
+              handleSubmit(values, setSubmitting)
+            }
+          >
+            {({ errors, handleChange, touched, submitForm, isSubmitting }) => (
+              <Form className={classes.form}>
+                <Grid container spacing={2}>
+                  <Grid item xs={12}>
+                    <TextField
```
Comment (andrewc910) [inline pages/users/signup.js:104]: If you pull in the import i said above, this email textfield can be replaced with:

```jsx
<Field
  component={TextField}
  variant="outlined"
  name="email"
  type="email"
  label="Email"
  value={values.email}
/>
```

Comment (andrewc910) [reply pages/users/signup.js:104]: All inputs can replaced with that. Errors will magically work.

File: pages/users/signup.js [hunk 5]
```diff
@@ -1,22 +1,171 @@
...
+    marginTop: theme.spacing(2),
+    color: 'grey',
+  },
+  login: {
+    margin: theme.spacing(3, 0, 3),
+  },
+}));
+
+const handleValidate = (values) => {
+  const errors = [];
```
Comment (andrewc910) [inline pages/users/signup.js:48]: Your validations aren't working because this should be an object not an array.

File: pages/users/signup.js [hunk 6]
```diff
@@ -1,22 +1,171 @@
...
+                      fullWidth
+                      onChange={handleChange}
+                      id="email"
+                      label="Email Address"
+                      name="email"
+                      helperText={
+                        errors.email && touched.email ? errors.email : null
+                      }
+                    />
+                  </Grid>
```
Comment (andrewc910) [inline pages/users/signup.js:116]: You don't need to close this grid and then open a new one. You can remove these two tags. Add a bottom-margin to each input so they don't touch.

File: pages/users/signup.js [hunk 7]
```diff
@@ -1,22 +1,171 @@
...
+                      label="Password"
+                      type="password"
+                      id="password"
+                      helperText={
+                        errors.password && touched.password
+                          ? errors.password
+                          : null
+                      }
+                    />
+                  </Grid>
```
Comment (andrewc910) [inline pages/users/signup.js:133]: Same as before: You don't need to close this grid and then open a new one. You can remove these two tags. Add a bottom-margin to each input so they don't touch.

File: pages/users/signup.js (hunk 2 above)
Comment (williampj) [reply pages/users/signup.js:100]: Yeah, I didn't know if we would use those things later

File: pages/users/signup.js (hunk 4 above)
Comment (williampj) [reply pages/users/signup.js:104]: Got it, `fullWidth` is still needed though I see.

File: pages/users/signup.js (hunk 5 above)
Comment (williampj) [reply pages/users/signup.js:48]: This is what working late a night does to you:)

**Comment by andrewc910** [comment]
Approved. Good Job :)

**Pull request merged**